code/game/machinery/firealarm.dm

```diff
@@ -157,28 +158,46 @@
 	if(machine_stat & NOPOWER)
 		return
 
-	. += mutable_appearance(icon, "fire_overlay")
-	if(is_station_level(z))
-		. += mutable_appearance(icon, "fire_[SSsecurity_level.get_current_level_as_number()]")
-		. += emissive_appearance(icon, "fire_[SSsecurity_level.get_current_level_as_number()]", src, alpha = src.alpha)
-	else
-		. += mutable_appearance(icon, "fire_[SEC_LEVEL_GREEN]")
-		. += emissive_appearance(icon, "fire_[SEC_LEVEL_GREEN]", src, alpha = src.alpha)
-
-	if(!(my_area?.fire || LAZYLEN(my_area?.active_firelocks)))
-		if(my_area?.fire_detect) //If this is false, leave the green light missing. A good hint to anyone paying attention.
-			. += mutable_appearance(icon, "fire_off")
-			. += emissive_appearance(icon, "fire_off", src, alpha = src.alpha)
-	else if(obj_flags & EMAGGED)
-		. += mutable_appearance(icon, "fire_emagged")
-		. += emissive_appearance(icon, "fire_emagged", src, alpha = src.alpha)
-	else
-		. += mutable_appearance(icon, "fire_on")
-		. += emissive_appearance(icon, "fire_on", src, alpha = src.alpha)
+	if(panel_open)
+		return
 
-	if(!panel_open && my_area?.fire_detect && my_area?.fire) //It just looks horrible with the panel open
-		. += mutable_appearance(icon, "fire_detected")
-		. += emissive_appearance(icon, "fire_detected", src, alpha = src.alpha) //Pain
+	if(obj_flags & EMAGGED)
+		. += mutable_appearance(icon, "fire_emag")
+		. += emissive_appearance(icon, "fire_emag_e", src, alpha = src.alpha)
+		set_light(l_color = LIGHT_COLOR_BLUE)
+
+	else if(!(my_area?.fire || LAZYLEN(my_area?.active_firelocks)))
```

Comment:
whyd you turn this into a bunch of nested ifs? it was cleaner before

Reply:
So the overlays get applied in the right order instead of multiple on top of each other.